include/eve/detail/function/simd/x86/slide_left.hpp

```diff
@@ -0,0 +1,147 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/implementation.hpp>
+#include <eve/conditional.hpp>
+
+namespace eve::detail
+{
+  template<real_scalar_value T, typename N, x86_abi ABI, std::ptrdiff_t Shift>
+  EVE_FORCEINLINE wide<T,N,ABI> slide_left_ ( EVE_SUPPORTS(sse2_)
+                                            , wide<T,N,ABI> v, index_t<Shift>
+                                            ) noexcept
+  requires(Shift <= N::value)
+  {
+          if constexpr(Shift == 0)        return v;
+    else  if constexpr(Shift == N::value) return wide<T,N,ABI>{0};
+    else
+    {
+      if constexpr( std::same_as<ABI,x86_128_>)
+      {
+        /// Mask noises from smaller sized registers
+        if constexpr(N::value < expected_cardinal_v<T,x86_128_>)
+        {
+          wide<T,expected_cardinal_t<T,x86_128_>> w = v.storage();
+          w &= keep_first(N::value).mask(as(w)).bits();
+          v = w.storage();
+        }
```

Comment:
I don't think I understand this. This is FFFF ...00000? For 8 chars it will be top 8 elements, that are zero. And you clear all of the rest. No?

Reply:
I keep the first N element with respect to the endianess. This is masking the data in the _m128 but outside the logical size of the wide so when shifting occurs, no garbage data are pulled in.

FYI, this has been tested to be required.

Reply:
Now maybe I can express it in a better way I concede that to you.